### PR TITLE
CHEAP-494 Fixing previous Survey methods, adding new update and release Survey mutation

### PR DESCRIPTION
**Pull request by bholagabbar**

### CODE REVIEW

File: src/app/controllers/SurveyController.js [hunk 1]
```diff
@@ -31,6 +30,31 @@ class SurveyController {
         });
     }
 
+    /**
+     * Update a Survey and return the ID of the updated object if successful
+     * @param {string} id - The id of the survey to be modified
+     * @param {Object} survey - The Modified Survey Object
+     * @returns {Promise<any>}
+     */
+    update(id, survey){
```
Comment (ralph-dev) [inline src/app/controllers/SurveyController.js:39]: Is there some authentication for these update mutations? Like what happens if someone just tries to update everyone's survey but they aren't the vendor that owns it etc?

File: src/app/controllers/SurveyController.js [hunk 2]
```diff
@@ -62,7 +108,7 @@ class SurveyController {
     createSurveyResponse(survey_id, survey_response) {
         return new Promise((resolve, reject) => {
             let mutationString = `
-                mutation createSurveyResponse ($survey_id: String!, $survey_response: CreateSurveyResponseInput!) {
+                mutation createSurveyResponseMutation ($survey_id: String!, $survey_response: CreateSurveyResponseInput!) {
```
Comment (junzhengca) [inline src/app/controllers/SurveyController.js]: I just realized `createSurveyResponseMutation` is not needed here, simply remove, and make it `mutation ($survey_id: ..........)`

File: src/app/controllers/SurveyController.js [hunk 3]
```diff
@@ -53,6 +77,28 @@ class SurveyController {
...
 
+    /**
+     * Release a Survey
+     * @param {string} id - The id of the Survey Object
+     * @returns {Promise<any>} - The id of the Survey object
+     */
+    release(id){
+        return new Promise((resolve, reject) => {
+            let mutationString = `
+                mutation releaseSurveyMutation ($id: String!) {
```
Comment (junzhengca) [inline src/app/controllers/SurveyController.js]: Same goes here

File: src/app/controllers/SurveyController.js [hunk 4]
```diff
@@ -31,6 +30,31 @@ class SurveyController {
...
+    /**
+     * Update a Survey and return the ID of the updated object if successful
+     * @param {string} id - The id of the survey to be modified
+     * @param {Object} survey - The Modified Survey Object
+     * @returns {Promise<any>}
+     */
+    update(id, survey){
+        return new Promise((resolve, reject) => {
+            let mutationString = `
+                mutation updateSurveyMutation ($id: String!, $survey: UpdateSurveyInput!) {
```
Comment (junzhengca) [inline src/app/controllers/SurveyController.js]: Same goes here.

File: src/app/controllers/SurveyController.js [hunk 5]
```diff
@@ -8,21 +8,20 @@ class SurveyController {
...
      * @param {Object} survey - The Survey Object
      * @returns {Promise<any>} 
      */
-    create(vendor_id, survey){
+    create(survey){
         return new Promise((resolve, reject) => {
             let mutationString = `
-                mutation createSurveyMutation ($vendor_id: String!, $survey: CreateSurveyInput!) {
-                    createSurvey(vendor_id: $vendor_id, survey: $survey) {
+                mutation createSurveyMutation ($survey: CreateSurveyInput!) {
```
Comment (junzhengca) [inline src/app/controllers/SurveyController.js]: Same

**Review by junzhengca — CHANGES_REQUESTED** [review]

**Review by junzhengca — APPROVED** [review]

**Pull request merged**